### PR TITLE
Implement rStar MCTS inference PR #30

### PR DESCRIPTION
**Pull request by YanSong97**
PR #30 

More testing is needed at the current stage, but feel free to have a try locally. :)

### CODE REVIEW

File: envs/rstar/eval_src/toolkit_for_MATH/simple_answer_check.py
Comment (ziyuwan) [inline envs/rstar/eval_src/toolkit_for_MATH/simple_answer_check.py:1]: This checker is not the same as qwen-math-checker in envs/math

**Review by ziyuwan — COMMENTED** [review]
may need to compare math-evaluation code with `envs/math`

File: reason/guided_search/rstar.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    def draw_tree(self):
+
+        def display_tree(node):
+            print("|" + "-" * (node.depth * 4) + str(node))
+            for child in node.children:
+                display_tree(child)
+
+        print(f"\n---------Expanded Tree---------")
+        display_tree(self.root)
```
Comment (ziyuwan) [inline reason/guided_search/rstar.py:200]: this function is useful for both `RStartSearchTree` and `SearchTree`, consider moving into `SearchTree`

I can do this, what's your opinion @YanSong97

**Review by ziyuwan — APPROVED** [review]

File: reason/guided_search/rstar.py (hunk 1 above)
Comment (YanSong97) [reply reason/guided_search/rstar.py:200]: yeah looks good to me

**Comment by windks** [comment]
这里的rStar是不是没有实现论文中的discriminate功能？

**Pull request merged**